src/execCmd.ts

```diff
@@ -0,0 +1,210 @@
+/*
+ * Copyright (c) 2021, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { join } from 'path';
+import { inspect } from 'util';
+import { fs as fsCore } from '@salesforce/core';
+import { Duration, env, parseJson } from '@salesforce/kit';
+import { AnyJson, isNumber } from '@salesforce/ts-types';
+import Debug from 'debug';
+import * as shelljs from 'shelljs';
+import { ExecCallback, ExecOptions, ShellString } from 'shelljs';
+
+export interface ExecCmdOptions extends ExecOptions {
+  /**
+   * Throws if this exit code is not returned by the child process.
+   */
+  ensureExitCode?: number;
+}
+
+export interface ExecCmdResult {
+  /**
+   * Command output from the shell.
+   *
+   * @see https://www.npmjs.com/package/shelljs#execcommand--options--callback
+   */
+  shellOutput: ShellString;
+
+  /**
+   * Command output parsed as JSON, if `--json` param present.
+   */
+  jsonOutput?: AnyJson;
+
+  /**
+   * The JsonParseError if parsing failed.
+   */
+  jsonError?: Error;
+
+  /**
+   * Command execution duration.
+   */
+  execCmdDuration: Duration;
```

Comment:
Thinking of someone who wants to exec a command and do an asserting based on a Duration, we're asking them to import our type (at least) and use add kit to their tests to get it and then look into Duration to see how to get out the various values from it.  

Given that a lot of JS (timeout, interval, etc) is standardized on ms as a unit, what if we returned that?

Reply:
we can totally use duration inside the testkit

Reply:
I like the flexibility and convenience of Duration.  I went back and forth on it's use and export type as well but decided that the potential benefits of it were too great.  For example, someone could use this to do simple command benchmark testing and easily get a time in any unit without doing the calculations themselves (and potentially messing it up).

Reply:
I'll export Duration from the lib